### PR TITLE
feature/SOF-7069 feature: isolate codemirror concerns

### PR DESCRIPTION
**Pull request by VsevolodX**

### CODE REVIEW

File: src/other/codemirror/CodeMirror.tsx [hunk 1]
```diff
@@ -6,10 +6,26 @@ import { StreamLanguage } from "@codemirror/language";
...
+    const deafultError = {
+        from: 0,
+        to: doc.length,
+        message: "Test",
+        severity: "error",
+    } as Diagnostic;
+    errors.push(deafultError);
+
+    return errors as Diagnostic[];
+});
```
Comment (VsevolodX) [inline src/other/codemirror/CodeMirror.tsx]: A temporary linter with one static message for experiments.

File: src/other/codemirror/CodeMirror.tsx [hunk 2]
```diff
@@ -35,31 +34,71 @@ export interface CodeMirrorProps {
...
+        if (prevProps.content !== content) {
+            this.setState({ content: content || "" });
+        }
     }
 
     /*
      * editor - CodeMirror object https://uiwjs.github.io/react-codemirror/
      * viewUpdate - object containing the update to the editor tree structure
      */
-    handleContentChange(newContent: string, viewUpdate: { origin: string }) {
```
Comment (VsevolodX) [inline src/other/codemirror/CodeMirror.tsx]: This `viewUpdate: { origin: string }` doesn't exist in CodeMirror 6, and WebApp and tests work fine with this parameter removed

File: src/other/codemirror/CodeMirror.tsx [hunk 3]
```diff
@@ -35,54 +36,56 @@ export interface CodeMirrorProps {
...
+
+        return entity.content !== nextEntity.content || entity.checks !== nextEntity.checks;
     }
 
     handleContentChange(newContent: string) {
         const { isEditing } = this.state;
         const { updateContent } = this.props;
         if (isEditing && updateContent) updateContent(newContent);
-        this.setState({ content: newContent });
+        this.setState({ entity: { content: newContent } });
```
Comment (VsevolodX) [inline src/other/codemirror/CodeMirror.tsx]: Here the checks get overwritten and do not chow up. But if I add `...prevState.checks`, or `state.entity.checks` -- the whole codemirror rerenders moving cursor at the beginning. 
Don't want to get stuck on this

Comment (VsevolodX) [reply src/other/codemirror/CodeMirror.tsx]: {entity.content: newContent}

Comment (VsevolodX) [reply src/other/codemirror/CodeMirror.tsx]: Works!

File: src/other/codemirror/CodeMirror.tsx [hunk 4]
```diff
@@ -35,54 +36,56 @@ export interface CodeMirrorProps {
...
+        return entity.content !== nextEntity.content || entity.checks !== nextEntity.checks;
     }
 
     handleContentChange(newContent: string) {
-        const { isEditing } = this.state;
+        const { isEditing, entity } = this.state;
         const { updateContent } = this.props;
         if (isEditing && updateContent) updateContent(newContent);
-        this.setState({ content: newContent });
+        this.setState({ ...entity, content: newContent });
```
Comment (timurbazhirov) [inline src/other/codemirror/CodeMirror.tsx]: 1. This has to be using `_resetStateEntity`
2. You are setting state directly, not its `entity` key

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**